### PR TITLE
Return Flash mode off manually. Because Google

### PR DESCRIPTION
**Pull request by Diolor**
Return Flash mode off manually when no flash instead of adding nullability in the parameters

### CODE REVIEW

File: fotoapparat/src/main/java/io/fotoapparat/parameter/Parameters.java [hunk 1]
```diff
@@ -29,7 +29,8 @@ public void putValue(Type type, Object value) {
 
     private void ensureType(Type type, Object value) {
         if (value == null) {
-            return;
+            throw new IllegalArgumentException("Provided " + type.clazz.getSimpleName()
```
Comment (dmitry-zaitsev) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/Parameters.java]: The idea behind `Parameters` is that it essentially behaves as a map and map does allow `null` values to be written. Moreover, we should always expect `null` values to be returned from `getValue` anyway, so this check does not really protects us from anything.

File: fotoapparat/src/main/java/io/fotoapparat/hardware/v1/capabilities/CapabilitiesFactory.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ public Capabilities fromParameters(Camera.Parameters parameters) {
 		List<String> supportedFlashModes = parameters.getSupportedFlashModes();
 		return supportedFlashModes != null
 				? supportedFlashModes
-				: Collections.<String>emptyList();
+				: Collections.singletonList(Camera.Parameters.FLASH_MODE_OFF);
```
Comment (dmitry-zaitsev) [inline fotoapparat/src/main/java/io/fotoapparat/hardware/v1/capabilities/CapabilitiesFactory.java]: I believe that this is the only change which is needed to fix the issue. I suggest we revert everything else.

Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/hardware/v1/capabilities/CapabilitiesFactory.java]: updated

**Comment by Diolor** [comment]
fixes #9

File: fotoapparat/src/main/java/io/fotoapparat/parameter/provider/InitialParametersValidator.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ */
+class InitialParametersValidator {
+
+    /**
+     * Validates a set of {@link Parameters} which the {@link io.fotoapparat.hardware.CameraDevice}
+     * needs.
+     *
+     * @param parameters The parameters to validate.
+     */
+    static void validate(Parameters parameters) {
```
Comment (dmitry-zaitsev) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/provider/InitialParametersValidator.java]: Lets make it not-static to make it properly object oriented.

File: fotoapparat/src/main/java/io/fotoapparat/parameter/provider/InitialParametersValidator.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        validateParameter(parameters, PICTURE_SIZE);
+        validateParameter(parameters, PREVIEW_SIZE);
+        validateParameter(parameters, FOCUS_MODE);
+        validateParameter(parameters, FLASH);
+    }
+
+    private static void validateParameter(Parameters parameters, Parameters.Type focusMode) {
```
Comment (dmitry-zaitsev) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/provider/InitialParametersValidator.java]: Update variable name. It is not necesserily a focus mode

File: fotoapparat/src/test/java/io/fotoapparat/parameter/provider/InitialParametersProviderTest.java [hunk 5]
```diff
@@ -129,4 +129,16 @@ public void selectPreviewSize_WithValidAspectRatio() throws Exception {
         );
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void initialParameterMissing() throws Exception {
```
Comment (dmitry-zaitsev) [inline fotoapparat/src/test/java/io/fotoapparat/parameter/provider/InitialParametersProviderTest.java]: This test tests validator and not provider. I suggest we make validator non-static and then just verify that it was called

**Pull request merged**